### PR TITLE
feat: extract reusable AOE answer validators

### PR DESCRIPTION
**Pull request by dmia-engineer**
Estimated review effort: 20-30 min

## Change Summary

> Jira: EU-1714  
> Extracted AOE answer validation into `ash-utils` as a reusable module under `ash_utils/aoe` and replaced monolithic validation with strategy-style per-type validators behind a common interface.  
> Added minimal typed input model (`AoeQuestionValidationInput`) using Pydantic `BaseModel` with only required fields (`ash_question_id`, `ash_question_type`, `validation_rules`, `options`).  
> Added dispatcher `AoeAnswerValidator` and dedicated validator classes for `number`, `text`, `boolean`, `date`, `datetime`, `select`, and `multi_select`.  
> Added comprehensive tests: dispatcher-level tests and dedicated class-based, parametrized validator tests (`autouse` fixture setup per class).  
> Bumped `ash-utils` version to `0.5.0` (`pyproject.toml`, `ash_utils/__init__.py`, `uv.lock`).  
> Backward compatibility: no breaking API changes for existing `ash-utils` modules; this is additive plus version bump.  
> Known risk: none identified in behavior regression; test coverage for new module is strong.

## How I know it works

- [x] I reviewed the code myself before requesting another engineer's review

> Verification steps:
> - Ran formatting/validation/tests in repo root:
>   - `UV_PYTHON=3.13 lets fmt`
>   - `UV_PYTHON=3.13 lets val`
>   - `UV_PYTHON=3.13 lets test`
> - Result: all checks passed (`108 passed`), pyright clean (`0 errors`).
> - Additional targeted runs:
>   - `UV_PYTHON=3.13 lets test tests/aoe/test_answer_validator.py`
>   - `UV_PYTHON=3.13 lets test tests/aoe/test_type_validators.py`

## Security

> No security impact.  
> No new endpoints, secrets, auth flows, or privilege changes introduced.

Made with [Cursor](https://cursor.com)

---

> [!NOTE]
> **Low Risk**
> Additive new validation/formatting utilities plus a version bump; existing behavior elsewhere is unaffected unless consumers adopt the new API.
> 
> **Overview**
> Adds a new `ash_utils.aoe` module that standardizes AOE answer validation and lab-formatting via a dispatcher (`AoeAnswerValidator`) and per-input-type strategy validators (`number`, `text`, `boolean`, `date`, `datetime`, `select`, `multi_select`).
> 
> Introduces a minimal, frozen Pydantic input model (`AoeQuestionValidationInput`) and exports the public API from `ash_utils.aoe.__init__`, with comprehensive unit tests covering dispatch and validator edge cases.
> 
> Bumps the package version to `0.5.0` in `ash_utils/__init__.py`, `pyproject.toml`, and `uv.lock`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 6e84cb18f0feab309dffadcb8ad394422aebc5af. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by dmia-engineer** [comment]
bugbot run

File: ash_utils/aoe/type_validators.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            raise ValueError(msg)
+        stripped_answer = answer.strip().lower()
+        if not stripped_answer:
+            msg = "at least one value is required"
+            raise ValueError(msg)
+        tokens = tuple(
+            token.strip().lower()
+            for token in stripped_answer.split(self.MULTI_SELECT_INPUT_SEPARATOR)
+            if token.strip().lower()
+        )
```
Comment (IlyaKaloshaAsh) [inline ash_utils/aoe/type_validators.py:107]: line already lowered before split. pls remove additional lower in for loop

**Review by IlyaKaloshaAsh — CHANGES_REQUESTED** [review]

**Review by IlyaKaloshaAsh — APPROVED** [review]

File: ash_utils/aoe/type_validators.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            return datetime.fromisoformat(answer)
+        except ValueError as exc:
+            msg = f"'{answer}' is not a valid RFC-3339 datetime"
+            raise ValueError(msg) from exc
+
+
+class SelectAoeAnswerTypeValidator(AoeAnswerTypeValidator):
+    def validate_and_format(self, question: AoeQuestionValidationInput, answer: str) -> str:
+        options = question.options
+        if not options:
```
Comment (rhys-ash) [inline ash_utils/aoe/type_validators.py:83]: low conviction: we're sort of mixing validation for the questions and validation for the answers here. these errors don't feel like they'd be actionable for customers at order placement time except for contacting ash support. in theory I'd imagine we'd do question validation ahead of time so we wouldn't run into this in practice though so NBD either way 🤷

**Review by rhys-ash — APPROVED** [review]
LGTM! Feel free to disregard the comment about validation except for maybe considering stronger messaging there since resolving AOE question issues would be outside of the client control?

File: ash_utils/aoe/type_validators.py (hunk 2 above)
Comment (dmia-engineer) [reply ash_utils/aoe/type_validators.py:83]: Yeah, good call out. I would raise a different type of error for this so we can differentiate.
Thanks, will fix it!

**Review by IlyaKaloshaAsh — APPROVED** [review]
